### PR TITLE
GET /api/articles (topic query)

### PR DESCRIPTION
**Pull request by Geno419**

### CODE REVIEW

File: __tests__/test.js [hunk 1]
```diff
@@ -230,16 +232,45 @@ describe("GET /api/users", () => {
...
+          article_id: expect.any(Number),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        };
+        res.body.articles.forEach((article) => {
+          expect(article).toMatchObject(expectedArticleStructure);
+        });
+      });
+  });
```
Comment (alice-y-codes) [inline __tests__/test.js:268]: Missing a test for Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper

File: db/model/model.js [hunk 2]
```diff
@@ -142,3 +127,28 @@ exports.fetchAllUsers = () => {
     )
     .then(({ rows }) => rows);
 };
+exports.getArticlesByTopic = (topic = "PassBYChecker") => {
+  let validTopicQueries = ["mitch", "cats", "PassBYChecker"];
+  if (!validTopicQueries.includes(topic)) {
+    return Promise.reject({ status: 400, msg: "Invalid sort_by query" });
+  }
+  validTopicQueries = ["mitch", "cats"];
```
Comment (alice-y-codes) [inline db/model/model.js:135]: This should be sorted via the db not hardcoded as your production data will be different. Please check if topic exists another way instead

**Review by alice-y-codes — COMMENTED** [review]
see comments
